### PR TITLE
Add tsubasa keyboard 

### PR DESCRIPTION
**Pull request by kuriatsu**
## Description

Initial PR for adding TSUBASA keyboard and keymaps (default and rotary encoder featured keymap)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/tsubasa/tsubasa.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "tsubasa.h"
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    return encoder_update_user(index, clockwise);
+}
```
Comment (zvecr) [inline keyboards/handwired/tsubasa/tsubasa.c:21]:
```suggestion
```

File: keyboards/handwired/tsubasa/keymaps/multifunc_encoder/keymap.c [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            } else {
+                // when released
+            }
+        }
+
+    }
+    return true;
+}
+
+bool should_process_keypress(void) { return true; }
```
Comment (zvecr) [inline keyboards/handwired/tsubasa/keymaps/multifunc_encoder/keymap.c:96]: Should use [sync options](https://docs.qmk.fm/#/feature_split_keyboard?id=data-sync-options) if required
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by kuriatsu** [comment]
Thank you for your review.
I wanted to use a rotary encoder and change the functions by pushing it in keymap/multifunc_encoder and LCD indicator on the slave halve.
I'm going to use `ENCODER_MAP_ENABLE` options, layers, and sync options instead of `switch` in keymap.c and `should_process_keypress`.
Then, remove `encoder_update_kb`.

**Comment by kuriatsu** [comment]
And for this PR, I will remove `keymap/multifunc_encoder` and follow your review.

File: keyboards/handwired/tsubasa/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# TSUBASA
+
+![tsubasa](https://github.com/kuriatsu/TSUBASA/blob/main/image/tsubasa.jpg)
```
Comment (drashna) [inline keyboards/handwired/tsubasa/readme.md:3]: Please rehost on Imgur.

File: keyboards/handwired/tsubasa/keymaps/multifunc_encoder/keymap.c (hunk 2 above)
Comment (drashna) [reply keyboards/handwired/tsubasa/keymaps/multifunc_encoder/keymap.c:96]: Specifically, keycodes shouldn't be processed on the secondary side, at all.  

If you need functionality to show up there, then the data sync options cover that, including custom synced data. 

This includes stuff like custom modes.

File: keyboards/handwired/tsubasa/readme.md (hunk 3 above)
Comment (kuriatsu) [reply keyboards/handwired/tsubasa/readme.md:3]: OK

**Comment by kuriatsu** [comment]
Thank you for your review.
I resolved the comments.

File: keyboards/handwired/tsubasa/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# TSUBASA
+
+![tsubasa](https://imgur.com/q5JlhvM)
```
Comment (drashna) [inline keyboards/handwired/tsubasa/readme.md:3]:
```suggestion
![tsubasa](https://i.imgur.com/q5JlhvMh.jpeg)
```

File: keyboards/handwired/tsubasa/rules.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes
+
+SPLIT_KEYBOARD = yes
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
```
Comment (drashna) [inline keyboards/handwired/tsubasa/rules.mk:24]: Encoder map should only be activated at the keymap level.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/tsubasa/readme.md (hunk 4 above)
Comment (kuriatsu) [reply keyboards/handwired/tsubasa/readme.md]: Oh, thank you.

File: keyboards/handwired/tsubasa/rules.mk (hunk 5 above)
Comment (kuriatsu) [reply keyboards/handwired/tsubasa/rules.mk]: I moved this option to the keymap level.

File: keyboards/handwired/tsubasa/config.h [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+#define RGB_DI_PIN D2
+#ifdef RGB_DI_PIN
+#    define RGBLED_NUM 12
+#    define RGBLIGHT_SPLIT
+#    define RGBLED_SPLIT {6, 6}
+#    define RGBLIGHT_HUE_STEP 8
+#    define RGBLIGHT_SAT_STEP 1
+#    define RGBLIGHT_VAL_STEP 1
+#    define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+// #    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/handwired/tsubasa/config.h:56]:
```suggestion
```

File: keyboards/handwired/tsubasa/readme.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+```bash
+qmk compile -kb handwired/tsubasa -km default
+```
+
+Flashing example for this keyboard:
+
+```bash
+qmk flash -kb handwired/tsubasa -km default
+```
```
Comment (drashna) [inline keyboards/handwired/tsubasa/readme.md:21]:
```suggestion
    make handwired/tsubasa:default

Flashing example for this keyboard:

    make handwired/tsubasa:default:flash
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request closed without merging**